### PR TITLE
docs(shell): add docs for shell library

### PR DESCRIPTION
**Pull request by zakariasahmane**
| Question         | Answer
| ---------------- | ---
| Branch?          |  feat/manager-shell
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| License          | BSD 3-Clause

## Description

Documentation for @ovh-ux/shell

### CODE REVIEW

**Comment by marie-j** [comment]
Maybe a `docs` commit would be more suitable in this case

File: packages/components/ovh-shell/README.md [hunk 1]
```diff
@@ -1 +1,105 @@
...
+  enablePlugin: () => enable(),
+  rename: (newName) => pluginRename(newName),
+};
+```
+
+- `setPluginAvailability`(`(pluginId: string, availability: boolean) => void`): Toggles plugin availability. If plugin isn't available, shell won't run it.
+  - `pluginId`: Your plugin `name`/`id`.
+  - `availabilty`: A boolean that sets the availability of a plugin to the desired value.
+
+### Plugin
```
Comment (antleblanc) [inline packages/components/ovh-shell/README.md:66]:
```suggestion
### Plugins
```

File: packages/components/ovh-shell/README.md [hunk 2]
```diff
@@ -1 +1,105 @@
...
+
+The goal of this lib is to provide utilities to communicate with a Shell. It exposes an API for that purpose, it also exposes a list of Plugins you can use with the Shell.
+
+## Table of contents
+
+- [Lib usage](#usage)
+- [API Docs](#api)
+  - [Shell methods](#shell)
+    - [initShell](#initShell)
+  - [Plugins API](#plugin)
```
Comment (antleblanc) [inline packages/components/ovh-shell/README.md:11]:
```suggestion
  - [Plugins API](#plugins)
```

File: packages/components/ovh-shell/README.md [hunk 3]
```diff
@@ -1 +1,105 @@
...
+  - `eventId`: The `name`/`id` of your event.
+  - `data`: Contains the data you want to send with your event. It can be everything that can be parsed by the navigator, objects such as callbacks or Promises won't work.
+
+- `registerPlugin`(`(pluginId: string, pluginApi: Record<string, Callback>) => void`): Register a plugin by giving its id and API.
+  - `pluginId`: Your plugin `name`/`id`.
+  - `pluginAPI`: This is an object that contains your plugin API. An API is basically just multiple callbacks indentified by a string.
+
+*Example of plugin API* :
+
+```js
```
Comment (antleblanc) [inline packages/components/ovh-shell/README.md:55]: For consistency:
```suggestion
```ts
```

File: packages/components/ovh-shell/README.md [hunk 4]
```diff
@@ -1 +1,105 @@
...
+- routing
+
+#### `i18n`
+
+- `getLocale`(`(shell: Shell, environment: Environment) => string`): Given a shell and an environment, returns the current locale of the user.
+- `setLocale`(`(locale: string) => void`): Sets new user locale based on given parameter.
+
+*Example :*
+
+```js
```
Comment (antleblanc) [inline packages/components/ovh-shell/README.md:81]: For consistency:
```suggestion
```ts
```

File: packages/components/ovh-shell/README.md [hunk 5]
```diff
@@ -1 +1,105 @@
...
+i18n.setLocale('en_GB');
+```
+
+#### `routing`
+
+- `initRouting`(`iframe: HTMLFrameElement`) : Initializes routing for the manager based on the URL in the iframe.
+
+*Example :*
+
+```js
```
Comment (antleblanc) [inline packages/components/ovh-shell/README.md:99]: For consistency:
```suggestion
```ts
```

**Review by antleblanc — APPROVED** [review]
Changes looks good to me!
I've left some comments, please let me know if you're OK with.
Thanks!

**Comment by zakariasahmane** [comment]
> Changes looks good to me! I've left some comments, please let me know if you're OK with. Thanks!

@antleblanc I'm okay with all the suggested changes, will apply them ! Thanks !

**Review by antleblanc — CHANGES_REQUESTED** [review]
Could you please resolve existing conflicts?
Thanks

**Review by antleblanc — CHANGES_REQUESTED** [review]
@zakariasahmane Could you please resolve existing conflicts?
Thank you!

**Pull request closed without merging**